hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/SqlConnector.java

```diff
@@ -355,28 +355,62 @@ default Vertex sinkProcessor(@Nonnull DagBuildContext context) {
 
     /**
      * Returns the supplier for the update processor that will update given
-     * {@code table}. The input to the processor will be the fields
-     * returned by {@link #getPrimaryKey(Table)}.
+     * {@code table}.
+     * <p>
+     * The processor works in 2 modes, depending on the value returned from
+     * {@link #dmlSupportsPredicates()}:<ol>
+     *
+     *     <li><b>With a predicate:</b> There will be no input to the processor.
+     *     The processor is supposed to update all rows matching the given
+     *     predicate. In this mode the `predicate` is always not-null.
+     *
+     *     <li><b>Without a predicate:</b> The processor is supposed to delete
+     *     all rows with primary keys it receives on the input. In this mode the
+     *     `predicate` is always null. The primary key fields are specified by
+     *     the {@link #getPrimaryKey(Table)} method.
+     *
+     * </ol>
      */
     @Nonnull
     default Vertex updateProcessor(
             @Nonnull DagBuildContext context,
             @Nonnull List<String> fieldNames,
-            @Nonnull List<HazelcastRexNode> expressions
+            @Nonnull List<HazelcastRexNode> expressions,
+            @Nullable HazelcastRexNode predicate
```

Comment:
If translation is not possible, what's then? For example: in MongoDB I may get some predicate, that cannot be translated into Mongo filters. I guess that user must manually add additional step with pulling the data (by as much as it's possible pushed down to source), then filter remaining data with all the filters that couldn't be pushed and then 3rd processor to actually perform the update - is that right?

Reply:
then the processor should generate two vertices: source + filter

Reply:
Does the engine really do a full table scan + filter, or does it take all the filters and pass them to the scan? I thought it was doing the latter.

Reply:
@frant-hartm The PR was updated last week. Now there's `SqlConnector.supportsExpression()` method, if the expression isn't supported, there will be scan+delete processors. If the expression is supported, there will be only delete processor and it will receive the predicate.